src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,114 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private static final double STEP_GROWTH = 1.5;
+    private int newArraySize = DEFAULT_SIZE;
+    private int size;
+    private T[] ownArrayList;
+
+    public ArrayList() {
+        ownArrayList = (T[]) new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= newArraySize) {
+            resizeArray();
+        }
+        ownArrayList[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        size++;
+        checkForValidIndex(index);
+        if (size >= newArraySize) {
+            resizeArray();
+        }
+        System.arraycopy(ownArrayList, index, ownArrayList, index + 1, size - index);
+        ownArrayList[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int indexOfAddedList = 0;
+        size += list.size();
+        for (int i = (size - list.size()); i < size; i++) {
+            ownArrayList[i] = list.get(indexOfAddedList);
+            indexOfAddedList++;
+        }
```

Comment:
You can reuse 'add(T value)' here
```suggestion
        for (int i = 0; i < list.size(); i++) {
            add(list.get(i));
        }
```